SpatialGDK/Source/SpatialGDKServices/Private/SpatialCommandUtils.cpp

```diff
@@ -496,4 +496,55 @@ bool SpatialCommandUtils::GetProcessInfoFromPort(int32 Port, FString& OutPid, FS
 	return false;
 }
 
+void SpatialCommandUtils::FetchRuntimeBinary(const FString& RuntimeVersion)
+{
+	FString RuntimePath = FPaths::Combine(SpatialGDKServicesConstants::GDKProgramPath, TEXT("runtime"), RuntimeVersion);
+	SpatialCommandUtils::GetPackage(RuntimeVersion, SpatialGDKServicesConstants::RuntimeExe, "runtime", RuntimePath, true);
+}
+
+void SpatialCommandUtils::FetchInspectorBinary(const FString& InspectorVersion)
+{
+	FString InspectorPath = FPaths::Combine(SpatialGDKServicesConstants::GDKProgramPath, TEXT("inspector"), InspectorVersion, SpatialGDKServicesConstants::InspectorExe);
+	SpatialCommandUtils::GetPackage(InspectorVersion, SpatialGDKServicesConstants::InspectorExe, "inspector", InspectorPath, false);
+}
+
+void SpatialCommandUtils::GetPackage(const FString& PackageVersion, const FString& PackageExe, const FString& PackageName, const FString& SaveLocation, const bool Unzip)
+{
+	FString PackagePath = FPaths::Combine(SpatialGDKServicesConstants::GDKProgramPath, *PackageName, PackageVersion);
+
+	// Check if the binary already exists for a given version
+	if (FPaths::FileExists(FPaths::Combine(PackagePath, PackageExe)))
+	{
+		UE_LOG(LogSpatialCommandUtils, Verbose, TEXT("%s binary already exists."), *PackageName);
+		return;
+	}
+
+	// If it does not exist then fetch the binary using `spatial worker package retrieve`
+	UE_LOG(LogSpatialCommandUtils, Log, TEXT("Trying to fetch %s version %s"), *PackageName, *PackageVersion);
+	FString Params = FString::Printf(TEXT("package retrieve %s %s %s %s"), *PackageName, *SpatialGDKServicesConstants::PlatformVersion,
+									 *PackageVersion, *SaveLocation);
+	if (Unzip)
+	{
+		Params += " --unzip";
+	}
+
+	TOptional<FMonitoredProcess> FetchingProcess;
+	const FString& ExePath = SpatialGDKServicesConstants::SpatialExe;
+	FetchingProcess = { ExePath, Params, true, true };
+	FetchingProcess->OnOutput().BindLambda([](const FString& Output) {
+		UE_LOG(LogSpatialPackageManager, Display, TEXT("Runtime: %s"), *Output);
+	});
+	FetchingProcess->Launch();
+
+	while (FetchingProcess->Update())
+	{
+		if (FetchingProcess->GetDuration().GetTotalSeconds() > SpatialGDKServicesConstants::ProcessTimeoutTime)
+		{
+			UE_LOG(LogSpatialPackageManager, Error, TEXT("Timed out waiting for the %s process fetching to start."), *PackageName);
+
+			FetchingProcess->Exit();
```

Comment:
This line replaces:
auto Handle = FetchingProcess->GetProcessHandle();
FPlatformProcess::TerminateProc(Handle);

I guess it is the same thing?

Reply:
Yup, we realised we didn't really need the KillProcess function at all